src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -173,14 +155,12 @@ private static LinqConstructorMapping BuildLinqConstructorMapping(MappingBuilder
         }
 
         // create a foreach loop with add calls if source is not an array
-        // and void ICollection.Add(T) or bool ISet.Add(T) is implemented and not explicit
-        // ensures .Add() is not called on immutable types
-        var hasImplicitCollectionAdd = ctx.Target.HasImplicitGenericImplementation(ctx.Types.Get(typeof(ICollection<>)), AddMethodName);
-        var hasImplicitSetAdd = ctx.Target.HasImplicitGenericImplementation(ctx.Types.Get(typeof(ISet<>)), AddMethodName);
-        if (ctx.Target.IsArrayType() || (!hasImplicitCollectionAdd && !hasImplicitSetAdd))
+        // and has an implicit add method
```

Comment:
IMO writing ".Add() method" would make this a little clearer

Would also spell out "for example"

Reply:
Improved.